src/help/components/landing/HelpQuickSearch.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  ChangeEvent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+  KeyboardEvent,
+} from 'react';
+import { generatePath, Link, useHistory, useLocation } from 'react-router-dom';
+import { Card, InfoList, SearchInput } from 'franklin-sites';
+import { debounce } from 'lodash-es';
+import qs from 'query-string';
+
+import CleanHighlightMarkDown from '../results/CleanHighlightMarkDown';
+
+import useDataApiWithStale from '../../../shared/hooks/useDataApiWithStale';
+
+import { help as helpURL } from '../../../shared/config/apiUrls';
+import { LocationToPath, Location } from '../../../app/config/urls';
+
+import { HelpSearchResponse } from '../../adapters/helpConverter';
+
+import styles from './styles/help-quick-search.module.scss';
+
+const numberResultsInView = 5 as const;
+
+type LocationState = { query: string };
+
+const HelpQuickSearch = () => {
+  const location = useLocation<undefined | LocationState>();
+  const history = useHistory();
+
+  const query = location?.state?.query || '';
+  const [searchValue, setSearchValue] = useState<string>(query);
+  const dataObject = useDataApiWithStale<HelpSearchResponse>(
+    query && helpURL.search({ query })
+  );
+  const replaceQueryInHistory = useMemo(
+    () =>
+      debounce((searchValue: string) => {
+        history.replace({
+          pathname: LocationToPath[Location.HelpResults],
+          state: {
+            query: searchValue || undefined,
+          },
+        });
+      }, 500),
+    [history]
+  );
+
+  useEffect(() => {
+    replaceQueryInHistory(searchValue);
+    return replaceQueryInHistory.cancel;
+  }, [replaceQueryInHistory, searchValue]);
+
+  const allArticlesLocation = useMemo(
+    () => ({
+      pathname: LocationToPath[Location.HelpResults],
+      search: qs.stringify({ query }),
+    }),
+    [query]
+  );
+
+  const handleKeyDown = useCallback(
+    ({ key }: KeyboardEvent<HTMLInputElement>) => {
+      if (key === 'Enter') {
+        history.push(allArticlesLocation);
+      }
+    },
+    [allArticlesLocation, history]
+  );
+
+  const allArticles = dataObject?.data?.results;
+  const infoData = allArticles
+    ?.slice(0, numberResultsInView)
+    .map(({ matches, title, id }) => {
+      const titleMatch = matches?.title?.[0];
```

Comment:
Are we sure there will always be at least 1 title, or should we check whether the array is empty? Same question for content.

Reply:
In the return of the map fn there is check for `titleMatch` and if it's not there just return the article title:
```
title: (
  <Link to={to}>
    {titleMatch ? <CleanHighlightMarkDown md={titleMatch} /> : title}
  </Link>
),
```
and in our API model we have `title` as required - could place additional checks if you think it's needed?

For the content we also check if there is a match but in this case it just returns undefined and the whole info list row isn't rendered. I can instead do:
```
content: contentMatch ? (
  <CleanHighlightMarkDown md={contentMatch} />
) : (
  ' '
),
```
so it will at least return the title like this (hardcoded to see what it would look like):
<img width="1091" alt="Screenshot 2021-09-10 at 12 07 34" src="https://user-images.githubusercontent.com/4357962/132845379-18ba7d79-9e5a-4d45-b29e-66c4c48f2001.png">